main.go

```diff
@@ -37,10 +38,11 @@ var (
 	sc = &SafeConfig{
 		C: &Config{},
 	}
-	configFile    = flag.String("config.file", "blackbox.yml", "Blackbox exporter configuration file.")
-	listenAddress = flag.String("web.listen-address", ":9115", "The address to listen on for HTTP requests.")
-	showVersion   = flag.Bool("version", false, "Print version information.")
-	timeoutOffset = flag.Float64("timeout-offset", 0.5, "Offset to subtract from timeout in seconds.")
+
+	// Flags
+	configFile    string
```

Comment:
I did initially, but it does not combine with the usage in the main Prometheus repo, where a Kingpin `application` is defined, and the flags need to be created with calls on this application instance.
I agree that pre-declaring the flags isn't very nice, which is what I referred to in the initial comment. To avoid it, we can either skip using the application (but diverge from how it is done in Prometheus), or move the variables into a config object, which allows us to exactly follow the Prometheus usage. But then I wonder if you'd prefer to merge it into the existing Config or SafeConfig types, or create a new one? This struct will need to be passed into the probeHandler func, as well.